pkg/initializer_v2/test/e2e/test_dataset.py

```diff
@@ -0,0 +1,107 @@
+import os
+import runpy
+import shutil
+import tempfile
+
+import pytest
+
+import pkg.initializer_v2.utils.utils as utils
+from sdk.python.kubeflow.storage_initializer.constants import VOLUME_PATH_DATASET
+
+
+class TestDatasetE2E:
+    """E2E tests for dataset initialization"""
+
+    @pytest.fixture(autouse=True)
+    def setup_teardown(self, monkeypatch):
+        """Setup and teardown for each test"""
+        # Create temporary directory for dataset downloads
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+        self.temp_dir = tempfile.mkdtemp(dir=current_dir)
+        os.environ[VOLUME_PATH_DATASET] = self.temp_dir
+
+        # Store original environment
+        self.original_env = dict(os.environ)
+
+        # Monkeypatch the constant in the module
+        import sdk.python.kubeflow.storage_initializer.constants as constants
+
+        monkeypatch.setattr(constants, "VOLUME_PATH_DATASET", self.temp_dir)
+
+        yield
+
+        # Cleanup
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+        os.environ.clear()
+        os.environ.update(self.original_env)
+
+    def verify_dataset_files(self, expected_files):
+        """Verify downloaded dataset files"""
+        if expected_files:
+            actual_files = set(os.listdir(self.temp_dir))
+            missing_files = set(expected_files) - actual_files
+            assert not missing_files, f"Missing expected files: {missing_files}"
+
+    @pytest.mark.parametrize(
+        "test_name, provider, test_case",
+        [
+            # Public HuggingFace dataset test
+            (
+                "HuggingFace - Public dataset",
+                "huggingface",
+                {
+                    "storage_uri": "hf://karpathy/tiny_shakespeare",
```

Comment:
Does anyone know which `dataset`/`model` in huggingface is suitable for the connectivity test?

Reply:
@seanlaii Which connectivity test do you want to perform ?

Reply:
I would like to test the actual downloading process and would like to know if there is any recommended dataset/model for testing. I currently choose a dataset that is only 1.11 MB.